js/piwik.js

```diff
@@ -3130,6 +3197,19 @@ if (typeof window.Matomo !== 'object') {
              * Browser features (plugins, resolution, cookies)
              */
             function detectBrowserFeatures() {
+                detectClientHints(function() {
+                    var i, requestType;
+                    clientHintsResolved = true;
+                    for (i = 0; i < clientHintsRequestQueue.length; i++) {
+                        requestType = typeof clientHintsRequestQueue[i];
+                        if (requestType === 'string') {
+                            sendRequest(clientHintsRequestQueue[i], configTrackerPause);
+                        } else if (requestType === 'object') {
```

Comment:
Maybe I am misunderstanding this block, but would that mean that matomo.js is sending more requests than it would before?

Reply:
Thanks, I forgot that the client hints check is asynchronous.